kore/src/Kore/Builtin/Map.hs

```diff
@@ -661,24 +667,40 @@ unifyEquals
             sequence (Map.intersectionWith unifyEqualsChildren map1 map2)
         let
             result
-              | not (Map.null remainder1) =
+              | not (Map.null remainder1) = do
                 -- There is nothing with which to unify the
                 -- remainder of map1.
-                ExpandedPattern.bottom
-              | not (Map.null remainder2) =
-                -- There is nothing with which to unify the
-                -- remainder of map2.
-                ExpandedPattern.bottom
+                Monad.Unify.explainBottom
```

Comment:
I think all of the unification failures here could be summarized as, "Cannot unify maps with different sizes". Extracting the cases would make it easier to scan the unifier.